cli/src/cluster_query.rs

```diff
@@ -1693,39 +1692,51 @@ pub fn process_show_stakes(
-            encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
-            ..RpcAccountInfoConfig::default()
-        },
-        ..RpcProgramAccountsConfig::default()
+    let mut pubkeys = HashSet::new();
+    if let Some(vote_account_pubkeys) = vote_account_pubkeys {
+        for key in vote_account_pubkeys {
+            pubkeys.insert(key.to_string());
+        }
     };
 
-    if let Some(vote_account_pubkeys) = vote_account_pubkeys {
-        // Use server-side filtering if only one vote account is provided
-        if vote_account_pubkeys.len() == 1 {
-            program_accounts_config.filters = Some(vec![
-                // Filter by `StakeState::Stake(_, _)`
-                rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
-                    offset: 0,
-                    bytes: rpc_filter::MemcmpEncodedBytes::Base58(
-                        bs58::encode([2, 0, 0, 0]).into_string(),
-                    ),
-                    encoding: Some(rpc_filter::MemcmpEncoding::Binary),
-                }),
-                // Filter by `Delegation::voter_pubkey`, which begins at byte offset 124
-                rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
-                    offset: 124,
-                    bytes: rpc_filter::MemcmpEncodedBytes::Base58(
-                        vote_account_pubkeys[0].to_string(),
-                    ),
-                    encoding: Some(rpc_filter::MemcmpEncoding::Binary),
-                }),
-            ]);
+    let vote_accounts = rpc_client.get_vote_accounts()?;
+    let mut vote_pubkeys = HashMap::new();
+    for current in &vote_accounts.current {
+        if pubkeys.contains(&current.vote_pubkey.to_string())
+            || pubkeys.contains(&current.node_pubkey.to_string())
+        {
+            vote_pubkeys.insert(
+                current.vote_pubkey.to_string(),
+                current.node_pubkey.to_string(),
+            );
+        }
+    }
+
+    for delinquent in &vote_accounts.delinquent {
+        if pubkeys.contains(&delinquent.vote_pubkey.to_string())
+            || pubkeys.contains(&delinquent.node_pubkey.to_string())
+        {
+            vote_pubkeys.insert(
+                delinquent.vote_pubkey.to_string(),
+                delinquent.node_pubkey.to_string(),
+            );
         }
     }
-    let all_stake_accounts = rpc_client
-        .get_program_accounts_with_config(&stake::program::id(), program_accounts_config)?;
+
+    // Requested number of keys does not match with found corresponding vote pubkeys
+    if pubkeys.len() != vote_pubkeys.len() {
+        for (key, value) in vote_pubkeys.iter() {
+            pubkeys.remove(key);
+            pubkeys.remove(value);
+        }
+        return Err(format!(
+            "Failed to retrieve some or all stake accounts: {:?}.",
+            pubkeys
+        )
+        .into());
+    }
+
+    let all_stake_accounts = rpc_client.get_program_accounts(&stake::program::id())?;
```

Comment:
Dropping the server-side filter when only one stake account is requested is not great for user and RPC server performance.  Any way to bring that back here?

Reply:
sure, i can take a look and see if i can modify the filter to allow for `node_pubkey`/`identity` and not just `vote_pubkey`